### ISSUE TITLE
[java] ImplicitFunctionalInterface should not be reported on sealed interfaces

### ISSUE THREAD

**Issue by marcindabrowski** [pmd/pmd#5724]
**Affects PMD Version:**  7.13.0

**Rule:** [ImplicitFunctionalInterface](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#implicitfunctionalinterface)

**Description:**

I have sealed interface:

```java
public sealed interface MyInterface permits MyInterface1, MyInterface2 {
  String message();
}
```

and PMD is reporting false positive for it with: `ImplicitFunctionalInterface:	Annotate this interface with @FunctionalInterface or with @SuppressWarnings("PMD.ImplicitFunctionalInterface") to clarify your intent.`

Sealed interfaces can't be annotated with `FunctionalInterface`.
